identity/identity.go

```diff
@@ -160,12 +170,12 @@ func (i *Identity) CopyWithoutCredentials() *Identity {
 	return &ii
 }
 
-func NewIdentity(traitsSchemaURL string) *Identity {
+func NewIdentity(traitsSchemaID uuid.UUID) *Identity {
 	return &Identity{
-		ID:              x.NewUUID(),
-		Credentials:     map[CredentialsType]Credentials{},
-		Traits:          Traits(json.RawMessage("{}")),
-		TraitsSchemaURL: traitsSchemaURL,
-		l:               new(sync.RWMutex),
+		ID:             x.NewUUID(),
+		Credentials:    map[CredentialsType]Credentials{},
+		Traits:         Traits(json.RawMessage("{}")),
+		TraitsSchemaID: traitsSchemaID,
```

Comment:
inject url if id is not empty - or make `SchemaURL` so that it returns an empty string if uuid is nil